src/platform/silabs/DiagnosticDataProviderImpl.cpp

```diff
@@ -82,31 +83,24 @@ CHIP_ERROR DiagnosticDataProviderImpl::ResetWatermarks()
 
 CHIP_ERROR DiagnosticDataProviderImpl::GetThreadMetrics(ThreadMetrics ** threadMetricsOut)
 {
-    /* Obtain all available task information */
-    TaskStatus_t * taskStatusArray;
     ThreadMetrics * head = nullptr;
-    uint32_t arraySize, x, dummy;
 
-    arraySize = uxTaskGetNumberOfTasks();
+    uint32_t threadCount         = osThreadGetCount();
+    osThreadId_t * threadIdTable = static_cast<osThreadId_t *>(chip::Platform::MemoryCalloc(threadCount, sizeof(osThreadId_t)));
 
-    taskStatusArray = static_cast<TaskStatus_t *>(chip::Platform::MemoryCalloc(arraySize, sizeof(TaskStatus_t)));
-
-    if (taskStatusArray != NULL)
+    if (threadIdTable != NULL)
     {
-        /* Generate raw status information about each task. */
-        arraySize = uxTaskGetSystemState(taskStatusArray, arraySize, &dummy);
-        /* For each populated position in the taskStatusArray array,
-           format the raw data as human readable ASCII data. */
-
-        for (x = 0; x < arraySize; x++)
+        osThreadEnumerate(threadIdTable, threadCount);
+        for (uint8_t tIdIndex = 0; tIdIndex < threadCount; tIdIndex++)
         {
+            osThreadId_t tId       = threadIdTable[tIdIndex];
             ThreadMetrics * thread = new ThreadMetrics();
```

Comment:
Do we know where this gets deallocated?

Reply:
yes in `DiagnosticDataProviderImpl::ReleaseThreadMetrics(ThreadMetrics * threadMetrics)`